### ISSUE TITLE
Deprecate use of jquery from Django Address

### ISSUE THREAD

**Issue by banagale** [furious-luke/django-address#139]
I've come to believe jquery should be eliminated from new software releases where possible. 

`django-address` uses jquery in the javascript functions that perform the auto complete function, ([jquery.geocomplete.js](https://github.com/furious-luke/django-address/blob/master/address/static/js/jquery.geocomplete.js)).

Does anyone have any strong feelings about keeping jquery in for any reason? Any chance someone is interested in taking on building a PR that finds a replacement or refactors the above library into pure ES 2015 plus vanilla JS?

**Comment by abmohan** [comment]
Good call, @banagale. Vanilla JS should be able to handle most of what's going on here. I wouldn't mind taking a stab at it.

**Comment by banagale** [comment]
Okay @abmohan Please do post if you begin work on this / any status updates you're able to offer. I'll assign this to you for now, if it looks like it can't get done just let me know.

**Comment by banagale** [comment]
Google offers its [own Places Autocomplete Javascript library](https://developers.google.com/maps/documentation/javascript/places-autocomplete#add-autocomplete) that handles the autocomplete feature on text inputs. 

Should be possible to update [`AddressWidget`](https://github.com/furious-luke/django-address/blob/2b77440ad27fb11d1e69db53193731abb49f741e/address/widgets.py#L19) to excise jquery.

**Comment by furious-luke** [comment]
Hey @banagale! I've got some time at the moment, I'll take a peek at how hard it might be to get Google's autocomplete library used here.
